addons/account/models/account_move.py

```diff
@@ -4819,3 +4820,9 @@ def get_invoice_localisation_fields_required_to_invoice(self, country_id):
         :return: an array of ir.model.fields for which the user should provide values.
         """
         return []
+
+    @api.onchange('invoice_payment_term_id')
```

Comment:
I didn't see that this is not where it should be :facepalm: 
There are sections in the file for each type of method...

Reply:
ouch, sorry ...